server/channels/store/sqlstore/shared_channel_store_membership.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package sqlstore
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+	sq "github.com/mattermost/squirrel"
+	"github.com/pkg/errors"
+)
+
+// UpdateRemoteMembershipCursor updates the LastMembersSyncAt timestamp for the specified SharedChannelRemote,
+// but only if the new timestamp is greater than the current value.
+func (s SqlSharedChannelStore) UpdateRemoteMembershipCursor(id string, syncTime int64) error {
+	query := s.getQueryBuilder().
+		Update("SharedChannelRemotes")
+
+	// Different syntax for MySQL vs PostgreSQL for conditional updates
+	if s.DriverName() == model.DatabaseDriverMysql {
+		query = query.
+			Set("LastMembersSyncAt", sq.Expr("GREATEST(LastMembersSyncAt, ?)", syncTime))
+	} else {
+		query = query.
+			Set("lastmemberssyncat", sq.Expr("GREATEST(lastmemberssyncat, ?)", syncTime))
+	}
+
+	query = query.Where(sq.Eq{"Id": id})
+
+	squery, args, err := query.ToSql()
+	if err != nil {
+		mlog.Error("Failed to generate SQL query",
```

Comment:
This is a common practice which we are trying to avoid. Which is double-logging. We are already returning the error, which will log it in the higher layers. But we are also logging it here. If there is not a huge benefit, I'd suggest to avoid this log statement here and in other places.

Reply:
cc @davidkrauser as well. We recently had the same discussion on another PR.